### PR TITLE
[SYCL] Add double GRF test

### PR DESCRIPTION
**Pull request by sarnex**
Corresponding test for https://github.com/intel/llvm/pull/6914 which is already merged

I manually verified this test passes on PVC

### CODE REVIEW

**Comment by sarnex** [comment]
@asudarsa Hey sorry do you mind reviewing this too? Just a system test that was requested in my other review. It's based on https://github.com/intel/llvm-test-suite/pull/1033

File: SYCL/DeviceCodeSplit/double-grf.cpp [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+//
+//===----------------------------------------------------------------------===//
+// This test verifies effect of
+//   set_kernel_properties(kernel_properties::use_double_grf);
+// API call in device code:
+// - ESIMD/SYCL splitting happens as usual
+// - SYCL module is further split into callgraphs for entry points requesting
+//   "double GRF" and callgraphs for entry points which are not
+// - SYCL device binary images requesting "double GRF" must be compiled with
+//   -doubleGRF option
```
Comment (asudarsa) [inline SYCL/DeviceCodeSplit/double-grf.cpp]: Is this the correct option?

Comment (sarnex) [reply SYCL/DeviceCodeSplit/double-grf.cpp]: good eye, it's technically right but it's not what we use anymore. fixed in latest commit

**Review by asudarsa — APPROVED** [review]
LGTM. Thanks

**Comment by sarnex** [comment]
@kbobrovs Hey Konst do you mind merging this? I don't have permission, thanks!

File: SYCL/DeviceCodeSplit/double-grf.cpp [hunk 2]
```diff
@@ -0,0 +1,139 @@
+//==----------- double_grf.cpp  - DPC++ SYCL on-device test ---------------==//
```
Comment (kbobrovs) [inline SYCL/DeviceCodeSplit/double-grf.cpp]:
```suggestion
//==----------- double-grf.cpp  - DPC++ SYCL on-device test ---------------==//
```

Comment (sarnex) [reply SYCL/DeviceCodeSplit/double-grf.cpp]: Fixed in latest commit, nice find!

**Review by kbobrovs — COMMENTED** [review]
LGTM with one minor note

**Review by kbobrovs — APPROVED** [review]

**Comment by kbobrovs** [comment]
> @kbobrovs Hey Konst do you mind merging this? I don't have permission, thanks!

Nick, please ping me once the testing passes.

**Comment by sarnex** [comment]
@kbobrovs Looks like testing passed, mind merging? Thanks

**Comment by kbobrovs** [comment]
@sarnex - sure, done

**Pull request merged**